File: Leetcode/1221. Split a String in Balanced Strings/solution.go
func balancedStringSplit(s string) int {
    count := 0
    res := 0
    
    for _, char := range s {
        if char == 'R' {
            count += 1
        } else {
            count -= 1
        }
        
        if count == 0 {
            res += 1
        }
    }
    
    return res
}

File: Leetcode/1221. Split a String in Balanced Strings/solution.java
class Solution {
    public int balancedStringSplit(String s) {
        int count = 0;
        int res = 0;
        
        for(int i =0; i < s.length(); i++){
            if(s.charAt(i) == 'R') {
                count++;
            } else{
                count--;
            }
            
            if(count == 0){ res++; }
        }
        
        return res;
    }
}